RemotePageProxy should send messages using AuxiliaryProcessProxy's PendingMessage queue

achristensen07 opened the pull request:
#### 8f046fe3a694e027782de5873b635e4403941507
<pre>
RemotePageProxy should send messages using AuxiliaryProcessProxy&apos;s PendingMessage queue
<a href="https://bugs.webkit.org/show_bug.cgi?id=259161">https://bugs.webkit.org/show_bug.cgi?id=259161</a>
rdar://112156432

Reviewed by J Pascoe.

Sometimes we try to send a message to a process while it is starting up before the connection
has been established.  Making RemotePageProxy no longer a MessageSender fixes this.
We now send messages using its process&apos;s send capabilities, which already has m_pendingMessages
to queue these messages and send them as soon as the process has a connection.

* Source/WebKit/UIProcess/RemotePageProxy.cpp:
(WebKit::RemotePageProxy::messageSenderConnection const): Deleted.
(WebKit::RemotePageProxy::messageSenderDestinationID const): Deleted.
* Source/WebKit/UIProcess/RemotePageProxy.h:
(WebKit::RemotePageProxy::send):
(WebKit::RemotePageProxy::sendWithAsyncReply):

Canonical link: <a href="https://commits.webkit.org/266017@main">https://commits.webkit.org/266017@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8f1a9c09249f5599f8ed8366c850be391640db51

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12577 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12906 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/13221 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/14317 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/12060 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/15408 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12924 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14751 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12741 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13520 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10642 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14747 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10798 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/18471 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11882 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11550 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14737 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12033 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9950 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11274 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15607 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1426 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/11889 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

pascoej commented on Source/WebKit/UIProcess/RemotePageProxy.h:
```diff
@@ -28,6 +28,7 @@
 #include "MessageReceiver.h"
 #include "MessageSender.h"
```
Is this include still required?

achristensen07 replied:
No!  Good catch!

achristensen07 committed:
RemotePageProxy should send messages using AuxiliaryProcessProxy's PendingMessage queue
https://bugs.webkit.org/show_bug.cgi?id=259161
rdar://112156432

Reviewed by J Pascoe.

Sometimes we try to send a message to a process while it is starting up before the connection
has been established.  Making RemotePageProxy no longer a MessageSender fixes this.
We now send messages using its process's send capabilities, which already has m_pendingMessages
to queue these messages and send them as soon as the process has a connection.

* Source/WebKit/UIProcess/RemotePageProxy.cpp:
(WebKit::RemotePageProxy::messageSenderConnection const): Deleted.
(WebKit::RemotePageProxy::messageSenderDestinationID const): Deleted.
* Source/WebKit/UIProcess/RemotePageProxy.h:
(WebKit::RemotePageProxy::send):
(WebKit::RemotePageProxy::sendWithAsyncReply):

Canonical link: https://commits.webkit.org/266017@main
